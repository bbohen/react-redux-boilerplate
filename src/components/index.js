export Button from './Button';
export Card from './Card';
export FormField from './FormField';
export Heading from './Heading';
export Modal from './Modal';
export NotFound from './NotFound';
