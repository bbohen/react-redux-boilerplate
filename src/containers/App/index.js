import styled from 'styled-components';

const App = styled.div`
  display: flex;
  flex-direction: column;
  font-size: 1em;
  background-color: ${props => props.theme.colors.background}
  min-height: 100vh;
`;

export default App;
