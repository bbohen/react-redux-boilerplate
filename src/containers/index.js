export App from './App';
export Cats from './Cats';
export ErrorBoundary from './ErrorBoundary';
export FormWithCats from './FormWithCats';
export Home from './Home';
export MainContent from './MainContent';
export Navigation from './Navigation';
