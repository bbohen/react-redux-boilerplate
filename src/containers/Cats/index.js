import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Card, Heading } from 'components';
import Wrapper from './Wrapper';
import { load as loadCats } from '../../redux/modules/cats';

export class UnconnectedCats extends Component {
  static propTypes = {
    catsAreLoaded: PropTypes.bool.isRequired,
    loadCats: PropTypes.func.isRequired,
    cats: PropTypes.arrayOf(PropTypes.shape),
  }

  static defaultProps = {
    cats: [],
  }

  componentWillMount() {
    const { catsAreLoaded } = this.props;

    if (!catsAreLoaded) {
      this.props.loadCats();
    }
  }

  render() {
    const { cats } = this.props;

    return (
      <Wrapper>
        {cats.length ? cats.map(({ data }, index) => (
          <Card
            index={index}
            key={data.id}
            imageUrl={data.thumbnail}
          />
        )) :
        <Heading>Loading you some cats</Heading>}
      </Wrapper>
    );
  }
}

export function mapStateToProps(state) {
  return {
    cats: state.cats.list,
    catsAreLoaded: state.cats.isLoaded,
  };
}

export default connect(mapStateToProps, { loadCats })(UnconnectedCats);
