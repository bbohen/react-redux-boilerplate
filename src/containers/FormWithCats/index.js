import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';

import { FormField } from 'components';
import { clearRandomCat, giveRandomCat } from '../../redux/modules/cats';
import validation from './validation';
import ActionText from './ActionText';
import CatFormHeading from './Heading';
import CatFormButton from './Button';
import Form from './Form';
import Wrapper from './Wrapper';

const UnconnectedFormWithCats = (props) => {
  const { email, number, required } = validation;
  let result;

  if (props.randomCat) {
    result = (
      <div>
        <CatFormHeading>
          Here is your cat.
          <ActionText onClick={props.clearRandomCat}>
            Try again?
          </ActionText>
        </CatFormHeading>
        <img
          alt=""
          src={props.randomCat}
        />
      </div>);
  }

  return (
    <Wrapper>
      {!result &&
      <Form onSubmit={props.handleSubmit && props.handleSubmit(props.giveRandomCat)}>
        <CatFormHeading>Form.</CatFormHeading>
        <Field
          label="Name:"
          name="name"
          component={FormField}
          type="text"
          validate={required}
        />
        <Field
          label="Email:"
          name="email"
          component={FormField}
          type="text"
          validate={[email, required]}
        />
        <Field
          label="Number:"
          name="number"
          component={FormField}
          type="text"
          validate={[number, required]}
        />
        <CatFormButton type="submit">
            Submit the form.
        </CatFormButton>
      </Form> }
      {result}
    </Wrapper>
  );
};

UnconnectedFormWithCats.propTypes = {
  clearRandomCat: PropTypes.func.isRequired,
  giveRandomCat: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  randomCat: PropTypes.string,
};

UnconnectedFormWithCats.defaultProps = {
  randomCat: '',
};

export default connect(
  ({ cats = {} }) => ({
    randomCat: cats.randomCat,
  }),
  {
    clearRandomCat,
    giveRandomCat,
  },
)(reduxForm({
  form: 'cats',
})(UnconnectedFormWithCats));
