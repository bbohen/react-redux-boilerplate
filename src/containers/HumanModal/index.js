import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import EnvelopeIcon from 'react-icons/lib/fa/envelope';
import BirthdayIcon from 'react-icons/lib/fa/birthday-cake';
import PhoneIcon from 'react-icons/lib/fa/phone';

import { Heading, Modal } from 'components';
import Details from './Details';
import Header from './Header';
import Icon from './Icon';
import Image from './Image';

const ModalHeaderHeading = styled(Heading)`
  margin-bottom: .5em;
`;

const HumanModal = ({ age, bio, birthday, imageUrl, email, name, phone, onClick, surname }) =>
  (<Modal onClick={onClick}>
    <Header>
      <Image
        alt=""
        src={imageUrl}
      />
      <Details>
        <ModalHeaderHeading>{name}, {surname}</ModalHeaderHeading>
        <div>
          <Icon>
            <EnvelopeIcon />
          </Icon>
          {email}
        </div>
        <div>
          <Icon>
            <BirthdayIcon />
          </Icon>
          {birthday.mdy} ({age} years old)
        </div>
        <div>
          <Icon>
            <PhoneIcon />
          </Icon>
          {phone}
        </div>
      </Details>
    </Header>
    <div>{bio}</div>
  </Modal>);

HumanModal.propTypes = {
  onClick: PropTypes.func.isRequired,
  age: PropTypes.number,
  bio: PropTypes.string,
  birthday: PropTypes.shape({
    mdy: PropTypes.string,
  }),
  email: PropTypes.string,
  imageUrl: PropTypes.string,
  name: PropTypes.string,
  phone: PropTypes.string,
  surname: PropTypes.string,
};

HumanModal.defaultProps = {
  age: 0,
  bio: '',
  birthday: {
    mdy: 'XX/XX/XXXX',
  },
  email: '',
  imageUrl: '',
  name: '',
  phone: '',
  surname: '',
};
export default HumanModal;
