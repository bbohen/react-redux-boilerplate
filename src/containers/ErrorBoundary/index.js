import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { App, MainContent } from 'containers';
import { Heading } from 'components';

const ErrorHeading = Heading.extend`
  margin-bottom: 40px;
`;

export default class ErrorBoundary extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      error: null,
      info: null,
      hasError: false,
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true, error, info });
  }

  render() {
    if (this.state.hasError) {
      return (
        <App>
          <MainContent>
            <div>
              <ErrorHeading>
              Whoops!
              </ErrorHeading>
              <div>{this.state.error.name}</div>
              <div>{this.state.error.message}</div>
              <div>{this.state.error.type}</div>
              <div>{this.state.error.arguments}</div>
            </div>
          </MainContent>
        </App>
      );
    }
    return this.props.children;
  }
}
