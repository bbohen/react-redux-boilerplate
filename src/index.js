import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import { MainContent, NotFound } from 'components';
import { App, Cats, Humans, FormWithCats, Home, Navigation } from 'containers';
import initializeStore from './redux/init';

import './globalStyles';

require('offline-plugin/runtime').install();

const store = initializeStore();

const Boilerplate = () => (
  <Provider store={store}>
    <BrowserRouter>
      <App>
        <Navigation />
        <MainContent>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/cats" component={Cats} />
            <Route path="/form" component={FormWithCats} />
            <Route path="/humans" component={Humans} />
            <Route component={NotFound} />
          </Switch>
        </MainContent>
      </App>
    </BrowserRouter>
  </Provider>
);

ReactDOM.render(
  <Boilerplate />,
  document.getElementById('mount'),
);

export default Boilerplate;
