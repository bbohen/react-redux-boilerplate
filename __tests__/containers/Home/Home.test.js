import React from 'react';

import { Home } from '../../../src/containers';
import testRendererWithProvider from '../../testUtils/testRendererWithProvider';

describe('<Home />', () => {
  it('renders the home container', () => {
    const component = testRendererWithProvider(<Home />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
