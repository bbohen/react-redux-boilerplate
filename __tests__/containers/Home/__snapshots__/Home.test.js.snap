// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders the home container 1`] = `
<div>
  <h2
    className="bhiNsp"
  >
    Hello
  </h2>
  <p>
    This is a boilerplate for react/redux apps. It grabs data from a couple of public APIs (reddit, giphy, uifaces) and does some random stuff.
  </p>
</div>
`;
